### DISCUSSION TITLE
Gatsby changing the layout of components.

### DISCUSSION THREAD

**Discussion by omar-shahid** [gatsbyjs/gatsby#31763] [Help] [answered]
## Description
First of all, thank you so much for making Gatsby! I am having this issue since the start of my gatsby journey, I thought it was from my side, but after long research, I couldn't find any solution. I have seen many issues regarding gatsby changing the production styles, maybe it is related to same bug. What gatsby is doing is replicating the wrapper element of any component, at any time, on any environment (i-e dev or production) randomly, I couldn't find any specific condition triggering this bug.

### What gatsby is actually rendering: (See the duplicate wrapper of text component)
![image](https://user-images.githubusercontent.com/40920737/120219425-ab2fa580-c254-11eb-9c98-334ea9fc4832.png)

### Steps to reproduce
Here's the link to repo:
[https://github.com/ReactReduxRouterDOM/public-gatsby-issue-repo](https://github.com/ReactReduxRouterDOM/public-gatsby-issue-repo)
- Install dependencies
- start development server
- try to refresh, navigate around pages.

### Expected result
Here are the screenshot of all the pages layout.

Home page /
![image](https://user-images.githubusercontent.com/40920737/120225066-04e89d80-c25e-11eb-9b9d-297709e4edf9.png)
404 page /404
![image](https://user-images.githubusercontent.com/40920737/120218746-a1f20900-c253-11eb-98a8-8608980d6fad.png)

Create lolly page /create-lolly
![image](https://user-images.githubusercontent.com/40920737/120218836-bd5d1400-c253-11eb-84bf-ef0603be6847.png)

### Actual result
![image](https://user-images.githubusercontent.com/40920737/120219425-ab2fa580-c254-11eb-9c98-334ea9fc4832.png)
![image](https://user-images.githubusercontent.com/40920737/120219621-f21d9b00-c254-11eb-9b1c-abe7a9ab511e.png)
![image](https://user-images.githubusercontent.com/40920737/120221687-470ee080-c258-11eb-9064-0fbe54a10026.png)

### Environment
Windows 10
Node: v14.16.1
NPM: 6.14.12
Yarn: 1.22.5
Gatsby CLI: 3.2.0
NPX: 6.14.12

Run `gatsby info --clipboard` in your project directory and paste the output here. Also name any `flags` you use inside `gatsby-config.js`.

```
  System:
    OS: Windows 10 10.0.19042
    CPU: (2) x64 Intel(R) Core(TM)2 Duo CPU     E7600  @ 3.06GHz
  Binaries:
    Node: 14.16.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.5 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.14.12 - C:\Program Files\nodejs\npm.CMD
  Languages:
    Python: 3.8.5
  Browsers:
    Edge: Spartan (44.19041.964.0), Chromium (90.0.818.66)

```

**Comment by LekoArts** [comment, marked as answer]
It's not a bug per se and also not specific to Gatsby as it can happen on every platform that generates HTML/CSS/JS and then rehydrates that content on the client (Rehydration). Somewhere in your code you could e.g. make the assumption that the `window` element exists and then during `gatsby build` that part will be `undefined` and your HTML will be different than what on the client will be rendered.

You could give this a read and check your code afterwards: https://www.joshwcomeau.com/react/the-perils-of-rehydration/

**Reply by omar-shahid** [reply]
That was exactly my issue. Thanks for linking this article! I think Gatsby should address this issue in their docs (kindly share the link if Gatsby already has a page) as this is related to rehydration,  so beginners like me don't create a new issue thinking it's a bug. Thanks again for solving my issue.
